lib/dock.rb

```diff
@@ -0,0 +1,41 @@
+
+require 'pry'
+class Dock
+  attr_reader :name, :max_rental_time, :rented_boat, :renters, :revenue
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = max_rental_time
+    @rented_boat = []
+    @renters = []
+    @revenue = 0
+
+  end
+
+
+
+  def rent(boat, renter)
+  @rented_boat << boat
+  @renters << renter
+  end
+
+
+  def log_hour
+    @rented_boat.each do |boat|
+     boat.add_hour
+   end
+    total_hours = @rented_boat.map do |boat|
+     boat.hours_rented
+   end
+
+    @revenue = @rented_boat.map do |boat|
+     boat.hours_rented * boat.price_per_hour
+   end.sum
+
+     if (total_hours.sum / 2 ) > 3
+       3
+     else
+       (total_hours.sum / 2)
+     end
+  end
```

Comment:
Seems like you moved forward trying to calculate the revenue before you incorporated the concept of returning a boat. Without that, any boat that has been rented will always end up being charged for the max rental time as long as we hit `log_hour` enough times. What happens if a boat is returned after two hours and three is the max rental time. Since we don't have any way to return a boat, it keeps being charged.

Also, I don't think that the `total_hours` can always be divided by two to give us an appropriate count of hours. That might work for a particular example, but don't think we can generalize.